extensions/websockets-next/server/runtime/src/main/java/io/quarkus/websockets/next/runtime/Mutex.java

```diff
@@ -0,0 +1,97 @@
+package io.quarkus.websockets.next.runtime;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.websockets.next.WebSocketServerConnection;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+
+/**
+ * TODO implement a more robust solution to ensure that endpoint callbacks are not invoked concurrently.
+ */
+class Mutex {
```

Comment:
I wonder if @franz1981 has anything ready to go that we could use here

Reply:
I was going to ask the very same question but not in this phase where we try to polish the API and functionality in general. You know the saying about premature optimization, right? ;-)

That said, @franz1981 if you have some recommendations I'm all ears :D